receiver/otelarrowreceiver/factory.go

```diff
@@ -0,0 +1,123 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package otelarrowreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/otelarrowreceiver"
+
+import (
+	"context"
+
+	"github.com/open-telemetry/otel-arrow/collector/sharedcomponent"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configgrpc"
+	"go.opentelemetry.io/collector/config/confignet"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver"
+)
+
+const (
+	typeStr = "otelarrow"
```

Comment:
you can use metadata.Type instead from the `generated internal/metadata/generated_status.go`.

Reply:
Done.